seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/commit/FileSinkAggregatedCommitter2.java

```diff
@@ -0,0 +1,120 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.file.sink.commit;
+
+import org.apache.seatunnel.api.sink.SinkAggregatedCommitter;
+import org.apache.seatunnel.connectors.seatunnel.file.sink.util.FileSystemUtils;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class FileSinkAggregatedCommitter2 implements SinkAggregatedCommitter<FileCommitInfo2, FileAggregatedCommitInfo2> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(FileSinkAggregatedCommitter2.class);
+
+    @Override
+    public List<FileAggregatedCommitInfo2> commit(List<FileAggregatedCommitInfo2> aggregatedCommitInfo) throws IOException {
+        List<FileAggregatedCommitInfo2> errorAggregatedCommitInfoList = new ArrayList<>();
+        aggregatedCommitInfo.forEach(aggregateCommitInfo -> {
+            try {
+                for (Map.Entry<String, Map<String, String>> entry : aggregateCommitInfo.getTransactionMap().entrySet()) {
+                    for (Map.Entry<String, String> mvFileEntry : entry.getValue().entrySet()) {
+                        // first rename temp file
+                        FileSystemUtils.renameFile(mvFileEntry.getKey(), mvFileEntry.getValue(), true);
+                    }
+                    // second delete transaction directory
+                    FileSystemUtils.deleteFile(entry.getKey());
+                }
+            } catch (Exception e) {
+                LOGGER.error("commit aggregateCommitInfo error ", e);
+                errorAggregatedCommitInfoList.add(aggregateCommitInfo);
+            }
+        });
+        return errorAggregatedCommitInfoList;
+    }
+
+    /**
+     * The logic about how to combine commit message.
+     *
+     * @param commitInfos The list of commit message.
+     * @return The commit message after combine.
+     */
+    @Override
+    public FileAggregatedCommitInfo2 combine(List<FileCommitInfo2> commitInfos) {
+        if (commitInfos == null || commitInfos.size() == 0) {
+            return null;
+        }
+        Map<String, Map<String, String>> aggregateCommitInfo = new HashMap<>();
+        Map<String, List<String>> partitionDirAndValuesMap = new HashMap<>();
+        commitInfos.forEach(commitInfo -> {
+            Map<String, String> needMoveFileMap = aggregateCommitInfo.computeIfAbsent(commitInfo.getTransactionDir(), k -> new HashMap<>());
+            needMoveFileMap.putAll(commitInfo.getNeedMoveFiles());
+            Set<Map.Entry<String, List<String>>> entries = commitInfo.getPartitionDirAndValuesMap().entrySet();
+            if (!CollectionUtils.isEmpty(entries)) {
```

Comment:
Want to check both null and empty? 
this call `getPartitionDirAndValuesMap().entrySet()` maybe throw an NPE

Or, use `commitInfo.getPartitionDirAndValuesMap() != null && !commitInfo.getPartitionDirAndValuesMap().isEmpty()`

Reply:
Good advice.